flytekit/types/structured/structured_dataset.py

```diff
@@ -390,6 +390,8 @@ def to_literal(
             )
 
         # Otherwise assume it's a dataframe instance. Wrap it with some defaults
+        if get_origin(python_type) is Annotated:
```

Comment:
should we have done this at the top of this function?  or is it okay here?

Reply:
we extract the python_type in [get_transformer](https://github.com/flyteorg/flytekit/blob/15e70b48b5ec862bdd1773eddd288769df45b6de/flytekit/core/type_engine.py#L530-L531) instead of [to_literal](https://github.com/flyteorg/flytekit/blob/15e70b48b5ec862bdd1773eddd288769df45b6de/flytekit/core/type_engine.py#L571-L581), so `python_type` could be `Annotated` here.